core/src/main/java/org/infinispan/interceptors/locking/AbstractLockingInterceptor.java

```diff
@@ -323,4 +325,14 @@ final Object nonTxLockAndInvokeNext(InvocationContext ctx, VisitableCommand comm
    private void handleUnlockAll(InvocationContext rCtx, VisitableCommand rCommand, Object rv, Throwable throwable) {
       lockManager.unlockAll(rCtx);
    }
+
+   private InvocationStage assertValidContext(KeyAwareLockPromise promise, InvocationContext ctx) {
```

Comment:
This is necessary to cover a corner case where the node handles the LockControlCommand concurrently with a DeadlockProbeCommand that rolls back a TX. Either, the LockControlCommand successfully acquires all the locks, which finishes the command execution and returns. Or, it identifies the TX rollback and finishes the command.

The problem was rolling back the TX *before* acquiring the locks. The locks would be acquired until the timeout elapses.